Main/Controllers/UsersController.cs

```diff
@@ -65,16 +74,14 @@ public IActionResult DeleteUser(string userId)
       if (user == null)
         return NotFound();
 
-      user.Collections.ToList().ForEach(c =>
+      foreach (var collection in user.Collections)
       {
-        _dbContext.SavedCollection.RemoveAll(sc => sc.CollectionId == c.Id);
-        _dbContext.SharedCollection.RemoveAll(sc => sc.CollectionId == c.Id);
-        _dbContext.Element.RemoveAll(el => el.CollectionId == c.Id);
-      });
-
-      _dbContext.Collection.RemoveAll(c => c.OwnerId == user.Id);
-      _userManager.DeleteAsync(user);
+        collection.IsPublic = false;
+      }
       _dbContext.SaveChanges();
+
+      await _userManager.SetLockoutEnabledAsync(user, true);
+      await _userManager.SetLockoutEndDateAsync(user, DateTimeOffset.Now.AddMonths(1));
```

Comment:
why 1 month?

Reply:
No time for proper UI so decided on a hardcoded option